### PR TITLE
feat: include encryption types with public key

### PR DESCRIPTION
**Pull request by Despire**
return a list of available encryption methods with the public key as described in #312

### CODE REVIEW

File: cmd/mailchain/internal/http/handlers/pubkey.go [hunk 1]
```diff
@@ -160,4 +166,10 @@ type GetPublicKeyResponseBody struct {
 	// Required: true
 	// example: hex/0x-prefix
 	PublicKeyEncoding string `json:"public_key_encoding"`
+
+	// Supported encryption methods for public keys.
+	//
+	// Required: true
+	// example: ["aes256cbc"]
```
Comment (robdefeo) [inline cmd/mailchain/internal/http/handlers/pubkey.go]: example: ["aes256cbc", "nacl", "noop"] is the full list

File: internal/pubkey/encryption_types.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/mailchain/mailchain/internal/protocols"
+)
+
+const (
+	// AES256CBC encryption type name.
+	AES256CBC = "aes256cbc"
+)
+
+// EncryptionMethods returns supported encryption methods.
+func EncryptionMethods(protocol string) ([]string, error) {
```
Comment (robdefeo) [inline internal/pubkey/encryption_types.go]: Encryption method that is allow is defined by the `PublicKey.Kind()` value. E.g.
- secpk2561 = [aes256cbc, noop]
- ed25519 = [nacl, noop]

Protocols usually support one key type but substrate actually supports mulitple

File: internal/pubkey/encryption_types.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	return []string{
+		AES256CBC,
+	}
+}
+
+func substrateEncryptionMethods() []string {
+	return []string{
+		AES256CBC,
+	}
+}
```
Comment (robdefeo) [inline internal/pubkey/encryption_types.go]: These can be removed based on above comment

**Comment by Despire** [comment]
blocked by #421

File: internal/pubkey/encryption_types.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+package pubkey
+
+import (
+	"fmt"
+
+	"github.com/mailchain/mailchain/crypto"
+)
+
+const (
```
Comment (robdefeo) [inline internal/pubkey/encryption_types.go]: These constants already exist in `crypto/cipher/encrypter/encrypter.go`, can you use these constants instead. Can you copy over your comments 😄

Comment (Despire) [reply internal/pubkey/encryption_types.go]: Thanks for pointing that out 😄 , didn't notice them.

**Pull request merged**